app/src/pages/inside/dashboardItemPage/newWidgets/charts/launchesComparisonChart/launchesComparisonChart.jsx

```diff
@@ -0,0 +1,277 @@
+import { TEST_ITEM_PAGE } from 'controllers/pages';
+import * as d3 from 'd3-selection';
+import { C3Chart } from '../../c3chart';
+import { renderTooltip } from '../../tooltip';
+import styles from './launchesComparisonChart.scss';
+import { chartConfigs } from '../../chartConfigs';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  statistics$executions$total: {
+    id: 'FilterNameById.statistics$executions$total',
+    defaultMessage: 'Total',
+  },
+  statistics$executions$passed: {
+    id: 'FilterNameById.statistics$executions$passed',
+    defaultMessage: 'Passed',
+  },
+  statistics$executions$failed: {
+    id: 'FilterNameById.statistics$executions$failed',
+    defaultMessage: 'Failed',
+  },
+  statistics$executions$skipped: {
+    id: 'FilterNameById.statistics$executions$skipped',
+    defaultMessage: 'Skipped',
+  },
+  statistics$defects$product_bug: {
+    id: 'FilterNameById.statistics$defects$product_bug',
+    defaultMessage: 'Product bug',
+  },
+  statistics$defects$automation_bug: {
+    id: 'FilterNameById.statistics$defects$automation_bug',
+    defaultMessage: 'Automation bug',
+  },
+  statistics$defects$system_issue: {
+    id: 'FilterNameById.statistics$defects$system_issue',
+    defaultMessage: 'System issue',
+  },
+  statistics$defects$no_defect: {
+    id: 'FilterNameById.statistics$defects$no_defect',
+    defaultMessage: 'No defect',
+  },
+  statistics$defects$to_investigate: {
+    id: 'FilterNameById.statistics$defects$to_investigate',
+    defaultMessage: 'To investigate',
+  },
+  ofTestCases: {
+    id: 'Widgets.ofTestCases',
+    defaultMessage: 'of test cases',
+  },
+});
+
+@injectIntl
+@connect(
+  (state) => ({
+    project: activeProjectSelector(state),
+  }),
+  {
+    redirect,
+  },
+)
+export class LaunchesComparisonChart extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    redirect: PropTypes.func.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
+    height: PropTypes.number,
+    project: PropTypes.string.isRequired,
+  };
+
+  static defaultProps = {
+    isPreview: false,
+    height: 0,
+  };
+
+  componentWillUnmount() {
+    this.node.removeEventListener('mousemove', this.getCoords);
+  }
+
+  onChartCreated = (chart, element) => {
+    this.chart = chart;
+    this.node = element;
+
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      return;
+    }
+
+    const legendClassName = cx('legend');
+    this.node.addEventListener('mousemove', this.getCoords);
+
+    d3
+      .select(this.node)
+      .insert('div', '.chart')
+      .attr('class', legendClassName)
+      .insert('div', `.${legendClassName}`)
+      .attr('data-js-legend-wrapper', '')
+      .selectAll('span')
+      .data(this.itemNames)
+      .enter()
+      .append('span')
+      .attr('data-id', (id) => id)
+      .attr('class', cx('legend-item'))
+      .html((id) => {
```

Comment:
Better to place this function as method of component started with `render*`, because it return some markup.